### PR TITLE
:sparkles: Allow using moid main

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

/test help

Testing

-  https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/3229

### CODE REVIEW

File: test/infrastructure/vcsim/controllers/envvar_controller.go [hunk 1]
```diff
@@ -287,14 +292,68 @@ func clusterEnvVarSpecGovmomiVariables(c *vcsimv1.ClusterEnvVarSpec) map[string]
...
+	if err != nil {
+		return vars, errors.Errorf("error creating test session: %v", err)
+	}
+
+	dc, err := s.Finder.Datacenter(ctx, vcsimhelpers.DatacenterName(datacenter))
+	if err != nil {
+		return vars, errors.Errorf("failed to locate datacenter reference: %v", err)
+	}
+
+	vars["VSPHERE_DATACENTER"] = dc.Reference().String()
```
Comment (dougm) [inline test/infrastructure/vcsim/controllers/envvar_controller.go:323]: The changed API doc says "name, inventory path or the MOID". But `Reference().String()` here (and so on below) is the MOR (Managed Object Reference). And `Reference().Value` is the MOID. The govmomi Finder changes support both, see: https://github.com/vmware/govmomi/blob/a2c22f0e33da798414846a99fd2f55522a405b13/object/common.go#L160-L166

Comment (rikatz) [reply test/infrastructure/vcsim/controllers/envvar_controller.go:323]: @dougm I will copy this comment and take it to kubernetes-sigs/cloud-api-provider-vsphere just to keep it on a single point :) this repo here is mostly to trigger the tests. But makes sense, I will change it to "support name, inventory path, moref or moid"

**Pull request closed without merging**